pkg/observ/observ.go

```diff
@@ -65,7 +65,14 @@ func Tracer(service string) buffalo.MiddlewareFunc {
 				requestAttrs(ctx.Request())...,
 			)
 
-			return next(&observabilityContext{Context: ctx, spanCtx: spanCtx})
+			handler := next(&observabilityContext{Context: ctx, spanCtx: spanCtx})
+
+			resp, ok := ctx.Response().(*buffalo.Response)
+			if ok {
+				span.SetStatus(trace.Status{Code: int32(resp.Status)})
```

Comment:
I don't think the http status and the OpenTrace status are mapped 1:1. You might want to use this: https://godoc.org/go.opencensus.io/plugin/ochttp#TraceStatus